### PR TITLE
[BugFix]fix bug when dropping a dead be, it reports "starletPort has not been updated by heartbeat from this backend"

### PR DESCRIPTION
**Pull request by abc982627271**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

image

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java [hunk 1]
```diff
@@ -395,11 +396,17 @@ public void dropBackend(String host, int heartbeatPort, boolean needCheckUnforce
             // remove worker
             if (Config.integrate_starmgr) {
                 long starletPort = droppedBackend.getStarletPort();
+                // only need remove worker when this be get dead after it became alive
+                // if it is dead all the time, not need to remove worker
+                boolean gotStarletPort = Deencapsulation.getField(droppedBackend, "gotStarletPort");
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java:401]: what if the worker is added with wrong starlet port, and its heartbeat is never sent to FE/StarManager, will it be easy to remove this worker anymore?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 8 (87.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/ComputeNode.java|2|3|66.67%|[393]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|5|5|100.00%|[]|

**Comment by abc982627271** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 21 / 26 (80.77%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|18|23|78.26%|[2011, 2012, 2015, 2018, 2021]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|3|3|100.00%|[]|

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**